Descrption: Testing ebuild patch that allows hardened-chromium and chromium-musl
patches to be placed in /etc/portage/patches/ for build testing.

Copyright (c) 2025 Aryan
SPDX-License-Identifier: BSD-3-Clause

Version: 1.0.0+testing

--- chromium.ebuid
+++ chromium.ebuid
@@ -49,7 +49,8 @@
 HOMEPAGE="https://www.chromium.org/"
 PPC64_HASH="7d1ac28278b5679d0b950ebd380bdd889b319592"
 PATCH_V="${PV%%\.*}-1"
-SRC_URI="https://chromium-tarballs.distfiles.gentoo.org/${P}-linux.tar.xz
+SRC_URI="https://commondatastorage.googleapis.com/chromium-browser-official/${P}.tar.xz -> ${P}-linux.tar.xz
+
 	!bundled-toolchain? (
 		https://gitlab.com/Matt.Jolly/chromium-patches/-/archive/${PATCH_V}/chromium-patches-${PATCH_V}.tar.bz2
 	)
@@ -1028,10 +1029,13 @@
 	# The sysroot is the oldest debian image that chromium supports, we don't need it
 	myconf_gn+=" use_sysroot=false"

-	# Use in-tree libc++ (buildtools/third_party/libc++ and buildtools/third_party/libc++abi)
-	# instead of the system C++ library for C++ standard library support.
-	# default: true, but let's be explicit (forced since 120 ; USE removed 127).
-	myconf_gn+=" use_custom_libcxx=true"
+	# Don't use in-tree libc++ for musl (buildtools/third_party/libc++ and buildtools/third_party/libc++abi),
+	# instead use system C++ library for C++ standard library support.
+        if use elibc_musl; then
+	    myconf_gn+=" use_custom_libcxx=false"
+        else
+            myconf_gn+=" use_custom_libcxx=true"
+        fi

 	# Disable pseudolocales, only used for testing
 	myconf_gn+=" enable_pseudolocales=false"
@@ -1185,6 +1189,15 @@
 		else
 			myconf_gn+=" is_cfi=${use_lto}"
 		fi
+
+		# Disable CFI on musl.
+		# https://github.com/12101111/overlay/issues/41.
+		if use elibc_musl; then
+			myconf_gn+=" is_cfi=false"
+		else
+			myconf_gn+=" is_cfi=${use_lto}"
+		fi
+
 		# Don't add symbols to build
 		myconf_gn+=" symbol_level=0"
 	fi
